enterprise-ws-spec/src/main/asciidoc/ServerProgrammingModel.adoc

```diff
@@ -351,295 +344,180 @@ web.xml or a partial web.xml:
 descriptor as a singleton session bean.
 
 The full requirements for Singleton Session Bean are defined in the
 section 4.8 of EJB 3.1 specification.
 
 [#anchor-46]
-===== Web container programming model
-
-====== Web container programming model for JAX-RPC
-
-The term JAX-RPC Service Endpoint used within the JAX-RPC specification
-is somewhat confusing since both Service Implementation Beans require
-the use of a JAX-RPC run time. However, in this case it refers to the
-programming model defined within the JAX-RPC specification that is used
-to create Web services that run within the web container. The
-requirements are repeated here with clarification. Changes from the
-JAX-RPC defined programming model are required for running in a Java EE
-container-managed environment.
-
-A JAX-RPC Service Endpoint can be single or multi-threaded. The
-concurrency requirement is declared as part of the programming model. A
-JAX-RPC Service Endpoint must implement javax.servlet.SingleThreadModel
-if single threaded access is required by the component. A container must
-serialize method requests for a Service Implementation Bean that
-implements the SingleThreadModel interface. Note, the SingleThreadModel
-interface has been deprecated in the Servlet 2.4 specification.
-
-The Service Implementation Bean must follow the Service Developer
-requirements outlined in the JAX-RPC specification and are listed below
-except as noted.
-
-* The Service Implementation Bean must have a default public
-constructor.
-
-* The Service Implementation Bean may implement the Service Endpoint
-Interface as defined by the JAX-RPC Servlet model. The bean must
-implement all the method signatures of the SEI. In addition, a Service
-Implementation Bean may be implemented that does not implement the SEI.
-This additional requirement provides the same SEI implementation
-flexibility as provided by EJB service endpoints. The business methods
-of the bean must be public and must not be static. If the Service
-Implementation Bean does not implement the SEI, the business methods
-must not be final. The Service Implementation Bean may implement other
-methods in addition to those defined by the SEI, but only the SEI
-methods are exposed to the client.
-
-* A Service Implementation must be a stateless object. A Service
-Implementation Bean must not save client specific state across method
-calls either within the bean instance’s data members or external to the
-instance. A container may use any bean instance to service a request.
-
-* The class must be public, must not be final and must not be abstract.
-* The class must not define the finalize() method.
-
-======= The optional ServiceLifecycle Interface
-
-A Service Implementation Bean for the web container may implement the
-java.xml.rpc.server.ServiceLifeCycle interface:
-
-package javax.xml.rpc.server;
-
-public interface *ServiceLifecycle *\{
+====== Web container programming model for Jakarta XML Web Services
```

Comment:
this heading can probably go away since there is no other model any more

Reply:
yes, will remove this heading.